### PR TITLE
Adds valid_pct support and TrimIntensity transform

### PR DESCRIPTION
**Pull request by s-mostafa-a**
I think I am not so good at naming!

### CODE REVIEW

**Comment by s-mostafa-a** [comment]
I think it is better to squash commits.

**Comment by jcreinhold** [comment]
Thanks for the update! Two things:

Regarding the `get_train_and_validation_from_one_directory` function:

1. Can you rename the function to `train_val_split` (like scikit-learn's [train_test_split](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html?highlight=split#sklearn.model_selection.train_test_split) function). The current name doesn't reflect what the function does (it says `one_directory` but it takes two directories as input.
2. Please change the docstring to use [Google style docstring](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#383-functions-and-methods). (See the style of the docstrings in the other classes/functions in the module.

Regarding the `normalize_hu` transform:

1. Is there a compelling reason to pigeonhole this function to Hounsfield units? Why not have a `TrimIntensity` transform? Which would take a min intensity, max intensity, and a new min and max. 

The code would look something like:

```python
class TrimIntensity:
    """ trim (and scale) the intensities of a sample """
    def __init__(self, min_val:float, max_val:float, 
                       new_min:float=0., new_max:float=1.,
                       tfm_x:bool=False, tfm_y:bool=True):
        self.min_val, self.max_val = min_val, max_val
        self.new_min, self.new_max = new_min, new_max
        self.tfm_x, self.tfm_y = tfm_x, tfm_y

    def _trim(self, x:torch.Tensor):
        x[x > self.max_val] = self.max_val
        x[x < self.min_val] = self.min_val
        return x

    def _scale(self, x:torch.Tensor):
        x = (x- self.min_val) / (self.max_val - self.min_val)
        x *= (self.new_max - self.new_min)
        x -= self.new_min
        return x

    def __call__(self, sample:Tuple[torch.Tensor, torch.Tensor]):
        src, tgt = sample
        if self.tfm_x:
            src = self._trim(src)
            src = self._scale(src)
        if self.tfm_y:
            tgt = self._trim(tgt)
            tgt = self._scale(tgt)
        return src, tgt
```

I haven't tested the above, so I don't know if that works, but something like that would be more general.

For both of these contributions, can you write unittests in the style of the tests in `test_dataset.py` (under the test directory). [For example](https://github.com/jcreinhold/niftidataset/blob/master/tests/test_dataset.py#L42). The unittest doesn't have to be particularly good (mine aren't) they just test that the dataset creates something of the right size and doesn't throw errors.

**Comment by s-mostafa-a** [comment]
Dear Jacob, Thank you for the points.
I updated the code with points you mentioned (except for unittest which I am going to add), but for TrimIntensity I think tfm_y would be not useful in this scenario. because inside the target and source tensors they are semantically different and cannot be trimmed with one (min_val, max_val). for the point you said, I think we should get params (min_val_src, max_val_src, min_val_tgt, max_val_tgt) inside constructor. (btw I think it still won't be useful this way)

**Comment by s-mostafa-a** [comment]
I am not happy yet with "train_val_split()" because it only works for NiftiDataset, not the other Dataset classes (e.g. MultimodalDataset, ...).

**Comment by s-mostafa-a** [comment]
Also I changed Normalize. The new feature (when mean and std are both None) will compute mean and std for each image and normalize it using them. As mentioned in 4.2 of this paper [V-NAS: Neural Architecture Search for Volumetric Medical Image Segmentation](https://arxiv.org/abs/1906.02817), it is useful to "decrease the data variance caused by the physical processes".

**Comment by jcreinhold** [comment]
Hi @s-mostafa-a thank you for the update. Please keep from adding new adjustments. Ideally, each addition and modification should be in their own PR. Unfortunately, I don't have the time to thoroughly review all of the changes. Below is a set of items to go through before I'll merge this PR.

1) Regarding the TrimIntensity transform, I agree that x and y can be semantically different, but they may not be. Consider the problem of image translation where x is an MR image and y is a CT image (for example). You may want to normalize the target because you only care about reconstructing the brain and don't care about the bone. While you are correct in assessing that the normalization/trim value for x will be different than for y, if you have the option to transform x and another option to transform y, you can instantiate two TrimIntensity classes and trim both, e.g.,
```python
trim_x = TrimIntensity(0, 1, tfm_x=True, tfm_y=False)
trim_y = TrimIntensity(-1, 1, tfm_x=False, tfm_y=True)
transforms = Compose([trim_x, trim_y])
```
2) Please use the existing test data to test your functions. It is already a mistake on my part to have binary test data embedded in this repo—adding more is compounding that problem.
3) Please make sure that the unittests only test one thing. Also, there is no reason to test for "no exception" in what you've written; if an exception is raised then the test failed, which is the purpose of the test.
4) Changing normalize to z-score normalize the input/outputs are fine. Please use pytorch mean and std to compute the mean and std, instead of first converting to numpy. I also don't see the use case of handling zero std with a kwarg. If the std is zero, then the image is constant; the mean will be equal to that constant which is subtracted out which creates an all zero image. If you want to handle the zero std case, just set it to 1e-6 and remove the kwarg to make the interface simpler.
5) Make sure that the style you are using matches the rest of the package, e.g., no spaces between kwargs/type annotations in function definitions (see the __init__ of NiftiDataset for an example). I understand that this is not PEP-8, but the style should be consistent throughout the package.
6) Please re-do your commits so that one commit affects one major modification/one addition, e.g., one commit adds the TrimIntensity module (and unittest), one commit adds the train_val_split (with unittest), etc. This will make reviewing the changes easier and better for future reference.
